docs/getting-started/apostrophe-cli.md

```diff
@@ -0,0 +1,163 @@
+
+**`--boilerplate`**
+
+Run `create-project` with a `--boilerplate` flag to start from a Github
+repository other than the standard `apostrophe-boilerplate` repo. For example,
+to create a project using the
+[Open Museum](https://github.com/apostrophecms/apostrophe-open-museum) demo:
+
+```bash
+apos create-project my-museum-clone --boilerplate=https://github.com/apostrophecms/apostrophe-open-museum.git
+```
+
+If you run the `create-project` command with the `--setup` flag, the command
+will also `npm install` the dependencies for the project and run
+`apostrophe-users:add` to create an admin user for the CMS, resulting in a fully
+ bootstrapped project. This command will prompt you for a password for the admin
+  user being created.
+
+## Create a piece type module
+To create the necessary files and basic configuration for a new Apostrophe piece
+type, run the following command from within your Apostrophe project's root
+directory:
+
+```bash
+apos create-piece vegetable
+```
+
+You must then register `'vegetables': {}` in `apps.js` in the `modules`
+object for it to start with the application.
+
+### options
+
+**`--pages`**
+
+If you run the `create-piece` command with the `--pages` flag, the command will
+also set up a corresponding [pieces-pages module](/core-concepts/reusable-content-pieces/browsing-directory-of-pieces) with your new piece type.
+
+```bash
+apos create-piece vegetable --pages
+```
+
+**`--widgets`**
+
+Similarly, you can run the `create-piece` command with the `--widgets` flag,
+which will also set up a corresponding [pieces-widgets module](/core-concepts/reusable-content-pieces/displaying-pieces-with-widgets.html#displaying-pieces-with-widgets) along with your new
+piece type. These flags can be used together or separately.
+
+```bash
+apos create-piece vegetable --widgets
+```
+
+So to create a piece type with both pages and widgets modules, you would run:
+
+```bash
+apos create-piece vegetable --pages --widgets
+```
+
+
+::: tip NOTE
+Be sure to use the *singular* version of the name of your content type. That
+will let the CLI attempt to set the piece type's `name`, `label`, and
+`pluralLabel` properties appropriately. The Apostrophe convention is for the
+`type` property of piece documents in the database to be singular (because they
+are a singular thing) and for the `name` of the piece *type* module to be plural
+(because it is used to create all the pieces of that type).
+
+As an example, this command:
+```bash
+apos create-piece dog
+```
+would generate this metadata:
+```javascript
+// lib/modules/index.js
+module.exports = {
+  name: 'dog',
+  extend: 'apostrophe-pieces',
+  label: 'Dog',
+  pluralLabel: 'Dogs',
+  addFields: []
+};
+```
+
+Do some words not pluralize using "s"? Yes, that's true. This
+function of the CLI is meant to help speed up your process in most cases, but in
+some cases it won't make sense. You can always manually change the module
+directory and those initial properties yourself.
```

Comment:
Grammer: "Do some words 'not'" is complicated. Why is this posed as a question at all? "It's true that some words don't pluralize this way.

Grammar: The subject here is a CLI feature, are we saying the feature 'won't make sense', or that the feature will make mistakes that you manually check.

Reply:
the functionality won't make sense, but it's built in so it'll make mistakes you manually check.